theano/configdefaults.py

```diff
@@ -1116,6 +1116,18 @@ def is_valid_check_preallocated_output_param(param):
              # Added elsewhere in the c key only when needed.
              in_c_key=False)
 
+if not config.cxx == "gcc" and config.gcc.cxxflags:
```

Comment:
clang++ is also suported. I think it should be:

    if config.cxx == "icpc" and config.gcc.cxxflags: